### PR TITLE
Move towards MySQL 8.0 as the default template generation

### PR DESCRIPTION
**Pull request by dbussink**
This upgrades the remaining things to Ubuntu 20.04 and makes MySQL 8.0 the default we run tests against. We still have tests for MySQL 5.7 but those are now explicitly opted into.

This should finish up the Ubuntu 20.04 upgrade and also makes things easier for the future when we need to upgrade again.

It is also marked as back port since we want to keep this updated on back ports as well or otherwise we're be blocked in the somewhat near future when 18.04 is deprecated. We probably want to look at changing the default maybe back there to 5.7 (to minimize other test changes that happened in previous PRs like https://github.com/vitessio/vitess/pull/11118) but that's a separate step to look at during back porting. 

## Related Issue(s)

Part of the upgrade already started with https://github.com/vitessio/vitess/pull/11026 but keeping things sensible in the CI generation scripts is much easier with this upgrade we think.

## Checklist

-   [x] "Backport me!" label has been added if this change should be backported
-   [x] Tests were added or are not required
-   [x] Documentation was added or is not required

### CODE REVIEW

File: test/ci_workflow_gen.go
Comment (deepthi) [inline test/ci_workflow_gen.go:120]: Line 117: do we still need this?

File: test/ci_workflow_gen.go
Comment (deepthi) [inline test/ci_workflow_gen.go:173]: This means that **two** workflow files are generated, one each for 5.7 and 8.0. This is why `vtorc_8.0` has been deleted from `clusterList`.

File: .github/workflows/cluster_endtoend_vtorc_mysql57.yml
Comment (deepthi) [inline .github/workflows/cluster_endtoend_vtorc_mysql57.yml:131]: This needs to be `vtorc_5.7` because we don't want to run the topology tests on 5.7

File: go/test/endtoend/vtorc/readtopologyinstance/main_test.go
Comment (GuptaManan100) [inline go/test/endtoend/vtorc/readtopologyinstance/main_test.go:96]: We can use `assert.Regexp(t, `[58]\.[07].*`, replicaInstance.Version)` instead. This will make this test work with both 5.7 and 8.0

**Review by GuptaManan100 — COMMENTED** [review]
Everything else looks good to me!

File: .github/workflows/cluster_endtoend_vtorc_mysql57.yml
Comment (GuptaManan100) [reply .github/workflows/cluster_endtoend_vtorc_mysql57.yml:131]: Yes, we would need to change this, or we can do what I suggest https://github.com/vitessio/vitess/pull/11153#discussion_r976281053 and run the topology tests for both versions and remove the vtorc_5.7 entirely from config.json

**Comment by deepthi** [comment]
I've addressed the feedback, and except for the known issue with vtbackup_transform, tests are passing.
When we merge this, we also need to remove a bunch of required checks and add new ones.
I'm inclined to wait until all of the WIP flag PRs are merged and then do one more merge from main into this PR before we merge it.

File: .github/workflows/cluster_endtoend_vtorc_mysql57.yml
Comment (deepthi) [reply .github/workflows/cluster_endtoend_vtorc_mysql57.yml]: Done

**Review by mattlord — APPROVED** [review]
❤️

**Pull request merged**
